### ISSUE TITLE
Rank Wikidata suggestions by popularity and type

### ISSUE THREAD

**Issue by brawer** [openstreetmap/iD#8437]
When iD proposes a list of Wikidata entries, consider ranking suggestions by popularity so that the most common things come first. Perhaps [QRank](https://qrank.toolforge.org/) might be useful for ranking (disclaimer: QRank author here, apologies for my shameless plug).

Also, consider type-specific boosting of entities in the Wikidata suggestion list: When editing animal enclosures, suggest animals before non-animals for `species:wikidata`. When editing flower beds and trees, suggest plants before non-plants. When editing fountains, suggest fountains (as per Wikidata “instance of”) for `wikidata` before non-fountains.

![image](https://user-images.githubusercontent.com/1527880/113259418-637cb400-92cd-11eb-9e3a-457117561b16.png)

**Comment by papac25** [comment]
- _![image](https://user-images.githubusercontent.com/39324597/113273029-3bc82480-92aa-11eb-9036-4da0309b1161.jpeg)_

**Comment by 1ec5** [comment]
> Perhaps QRank might be useful for ranking

Does QRank have an API associated with it? iD runs in a Web browser rather than on a desktop, so it would need some way of getting sorted results from an API on demand. Currently, iD uses the `wbsearchentities` action in Wikidata’s Wikibase API.

**Comment by matkoniecz** [comment]
@brawer Does QRank have an API associated with it?

**Comment by brawer** [comment]
> @brawer Does QRank have an API associated with it?

Not yet, but it would be easy and not that much work to build. If there's a concrete use case (such as the iD editor), we can build it.

To run in production, I would not recommend Wikimedia's datacenter: while they have a public-facing cloud (Wikimedia Toolforge, used for qrank.toolforge.org), Wikimedia's public cloud is not reliable enough for a real produciton server, such as an API server that gets queries from every iD user. (Wikimedia also operates high-quality infrastructure, for serving Wikipedia to the world, but that's entirely different from toolforge, and not open to external projects, so not an option here).But these days, there's lots of cheap, production-quality clouds out there, so this is not really a blocker.

@matkoniecz How many queries per second would iD send to an API, roughly? Does iD already operate any servers?

**Comment by matkoniecz** [comment]
> Does iD already operate any servers?

no, it is served (primarily) from OpenStreetMap Foundation servers as part of the osm.org website and uses general OpenStreetMap Foundation API/database server

> How many queries per second would iD send to an API, roughly?

that would trigger whenever user tries to add `wikidata`

on one hand some edit add multiple, but only small fraction of edit does

wild guess, overestimating it heavily would say as much as 10% of changesets

per https://github.com/openstreetmap/iD/issues/8307#issuecomment-2454204783 we 18k edits per day (less than I thought...)

so 1800 per day, that comes to  ((1800 / 24) / 60) / 60 = 0.02 per second

**Comment by tyrasd** [comment]
by now, we do include the wikidata descriptions in the dropdown, which helps a lot to disambiguate:

<img width="400" src="https://github.com/user-attachments/assets/179b67d7-f428-4c3b-a720-4ee270cc4d4a" />

And _order by popularity_ is something I would expect the wikidata search API to do for me, isn't it?

Anything else seems to be too much effort for what we need at the moment.
